### PR TITLE
[C-Api/Single] create in/out handle in single-shot

### PR DESCRIPTION
**Pull request by jaeyun-jung**
Use gst-info and create handle inside of single-shot instance.
Referencing struct does not initialize lock and other variables.

### CODE REVIEW

**Review by anyj0527 — APPROVED** [review]
LGTM

**Comment by jaeyun-jung** [comment]
@anyj0527 I updated this PR - create data handle in single instance and handle error case. Please review the changes.

**Review by songgot — APPROVED** [review]
LGTM

File: c/src/ml-api-inference-single.c [hunk 1]
```diff
@@ -1600,7 +1592,8 @@ ml_single_set_input_info (ml_single_h single, const ml_tensors_info_h info)
         "The parameter, info (const ml_tensors_info_h), is not valid. Although it is not NULL, the content of 'info' is invalid. If it is created by ml_tensors_info_create(), which creates an empty instance, it should be filled by users afterwards. Please check if 'info' has all elements filled with valid values.");
 
   ML_SINGLE_GET_VALID_HANDLE_LOCKED (single_h, single, 0);
-  status = ml_single_set_gst_info (single_h, info);
+  _ml_tensors_info_copy_from_ml (&gst_info, info);
+  status = ml_single_set_gst_info (single_h, &gst_info);
   ML_SINGLE_HANDLE_UNLOCK (single_h);
```
Comment (anyj0527) [inline c/src/ml-api-inference-single.c:1598]: may `gst_tensors_info_free (&gst_info)` be added.

Comment (jaeyun-jung) [reply c/src/ml-api-inference-single.c:1599]: Thanks, updated.

**Pull request merged**